web/src/contexts/AuthContext.tsx

```diff
@@ -1,33 +1,80 @@
-import { createContext, useState, useContext } from "react";
-import { getToken, saveToken, removeToken } from "../helpers/localStorage";
+import { createContext, useState, useContext, useEffect } from "react";
+import { saveToken, removeToken, Token } from "../helpers/localStorage";
+import { graphql, useMutation } from 'react-relay'
+
+import type { AuthContextMutation } from './__generated__/AuthContextMutation.graphql'
 
 type AuthContextProps = {
-  token: string,
-  signIn: (token: string) => void,
+  token: Token | undefined,
+  signIn: (token: Token) => void,
   signOut: () => void
 }
 
-export const AuthContext = createContext<AuthContextProps>({
-  token: '',
-  signIn: (token: string) => { },
-  signOut: () => { }
-})
+export const AuthContext = createContext<AuthContextProps | undefined>(undefined)
 
 export const AuthProvider = ({ children }: { children: React.ReactNode }) => {
-  const [token, setToken] = useState<string>(() => {
-    return getToken() || ''
-  })
+  const [token, setToken] = useState<Token | undefined>(undefined)
+
+  const [commitRefresh] = useMutation<AuthContextMutation>(graphql`
+    mutation AuthContextMutation($input: userRefreshTokenInput!) {
+      userRefreshToken(input: $input){
+        accessToken
+        error
+      }
+    }
+  `)
 
-  const signIn = (token: string) => {
+  const signIn = (token: Token) => {
     setToken(token)
     saveToken(token)
   }
 
   const signOut = () => {
-    setToken('')
+    setToken(undefined)
     removeToken()
   }
 
+  const everyTenMinutes = (callback: () => void): NodeJS.Timer => {
+
+    /* change --------------------------------------| 
+                                                    V this to 0.1 to test manually (yes) */
+    const TEN_MINUTES_IN_MILLISECONDS = 1000 * 60 * 10
+
+    return setInterval(callback, TEN_MINUTES_IN_MILLISECONDS)
+  }
+
+  const { refreshToken } = token || {}
+
+  useEffect(() => {
+    if (!refreshToken?.refreshToken) {
```

Comment:
`refreshToken?.refreshToken` doesn't looking good, try to think on a better name.